Feat: Whiteboards color system

sprocketc opened the pull request:
The implementation is similar to https://github.com/logseq/logseq/pull/6821
- Valid color strings (eg red) are stored
- Colors on whiteboards created with previous versions should still be compatible with the new system
- The default color value is an empty string (could be problematic if you try to open a whiteboard created with the new system with an older version of Logseq)
- The slider modifies the opacity of selected elements
- The default opacity on highlight component is now 0.5, and can be increased it up to 1
- Changed the icon of the fill toggle and removed the opacity from the toggle states

I used the `--ls-wb` prefix on the introduces css variables. I think we need to move away from the implementation specific `-tl`, but I would like to hear your thoughts on this. 

One of the things I also considered changing but decided not to, was the stroke and fill properties. Ideally we would like to have a single `color` property for all elements. The color should be translated into fill and/or stroke depending on the component. The problem with this approach is that it is going to be hard to maintain compatibility with previous versions, but it will allow us to avoid situations like [this one](https://github.com/logseq/logseq/pull/7111/files#diff-ccbda2a868b08935e9a14905ab6a0d3a49a479e4576806388d963d29f1f9628cL346).

![Screenshot from 2022-10-31 18-43-15](https://user-images.githubusercontent.com/10744960/199062224-adac65eb-2a95-4a98-9b96-96643d8ab5a4.png)

[Screencast 2 from 10-31-2022 06:38:46 PM.webm](https://user-images.githubusercontent.com/10744960/199061854-6d61a51b-9595-4b0e-8892-a759ef41694a.webm)

pengx17 reviewed (changes requested):
Impressive work!
I am ok with the `--ls-wb` prefix. Are you suggesting we should refactor existing class names to this on instead?
Just keep in mind we may want to provide these as a kind of non-documented APIs that allow the user to override them by using custom.css. We may not have many different conventions to confuse the users and if we do, we should do it earlier.

---

> One of the things I also considered changing but decided not to, was the stroke and fill properties. Ideally we would like to have a single color property for all elements. The color should be translated into fill and/or stroke depending on the component. The problem with this approach is that it is going to be hard to maintain compatibility with previous versions, but it will allow us to avoid situations like [this one](https://github.com/logseq/logseq/pull/7111/files#diff-ccbda2a868b08935e9a14905ab6a0d3a49a479e4576806388d963d29f1f9628cL346).

Yeah, it is really troublesome to maintain two color attributes. I think it is ok to do it right now and make a simple migration function in the deserialize function TLHistory.

---

**Below are some ux issues I found and may need attention from Jakob**

IMO this little opacity handle seems to be too low contrast with the background.
<img width="222" alt="image" src="https://user-images.githubusercontent.com/584378/199369246-691db42c-87bd-4c79-8fd1-c769876e82b3.png">

---

Is it possible to make the colors consistent? The color in swatch/shape and shape preview is too different.
<img width="966" alt="image" src="https://user-images.githubusercontent.com/584378/199369512-4f0ba388-1944-4264-88ff-a3b5df3b2437.png">

---

The panel of the swatch will be detached if moving the canvas.
<img width="672" alt="image" src="https://user-images.githubusercontent.com/584378/199370142-58d68088-8d91-4216-ab6e-cb9baa8520cf.png">

---

The portal becomes hardly to see the text contents when it gets a low opacity. Not sure if it is better to change the opacity of the background only, instead of changing the opacity of the whole shape?
<img width="594" alt="image" src="https://user-images.githubusercontent.com/584378/199370622-3a076c40-3b7e-4b58-8713-672e427e4394.png">

pengx17 commented on tldraw/apps/tldraw-logseq/src/lib/shapes/PenShape.tsx:
```diff
@@ -25,8 +25,8 @@ export class PenShape extends TLDrawShape<PenShapeProps> {
-    stroke: '#000000',
-    fill: '#ffffff',
+    stroke: '',
+    fill: 's',
```
Seems like a typo here

pengx17 commented:
I like your solution for the color system added to the whiteboard ♥️. There are some minor code styling issues that can be fixed by running prettier and unused imports, but feel free to do that in another pr.

sprocketc commented:
> Are you suggesting we should refactor existing class names to this on instead?

Yes, but I think this needs to be done on a separate enhancement if we decide to do this.

> IMO this little opacity handle seems to be too low contrast with the background.

Added a primary colored border.

> Is it possible to make the colors consistent? The color in swatch/shape and shape preview is too different.

The preview colors should now be fixed. The bg color on the swatches is a design decision.

> The panel of the swatch will be detached if moving the canvas.

Fixed.

> The portal becomes hardly to see the text contents when it gets a low opacity. Not sure if it is better to change the opacity of the background only, instead of changing the opacity of the whole shape?

Handled.

> There are some minor code styling issues that can be fixed by running prettier

Done.
